monai/metrics/cumulative_average.py

```diff
@@ -9,64 +9,162 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import torch
-
-from monai.transforms import isnan
-from monai.utils import convert_data_type
+from typing import Any, Union
 
-from .metric import Cumulative
+import torch
+import torch.distributed as dist
+from numpy.typing import NDArray
 
 
-class CumulativeAverage(Cumulative):
+class CumulativeAverage:
     """
-    Cumulatively record data value and aggregate for the average value.
-    It supports single class or multi-class data, for example,
-    value can be 0.44 (a loss value) or [0.3, 0.4] (metrics of two classes).
-    It also supports distributed data parallel, sync data when aggregating.
-    For example, recording loss values and compute the overall average value in every 5 iterations:
+    A utility class to keep track of average values. For example during training/validation loop,
+    we need to accumulate the per-batch metrics and calculate the final average value for the whole dataset.
+    When training in multi-gpu environment, with DistributedDataParallel, it will average across the processes.
+
+    Example:
 
     .. code-block:: python
 
-        average = CumulativeAverage()
-        for i, d in enumerate(dataloader):
-            loss = ...
-            average.append(loss)
-            if i % 5 == 0:
-                print(f"cumulative average of loss: {average.aggregate()}")
-        average.reset()
+        from monai.metrics import CumulativeAverage
+
+        run_avg = CumulativeAverage()
+        for i in range(len(train_set)):
+            ...
+            val = calc_metric(x,y) #some metric value
+            run_avg.append(val)
+
+        val_avg = run_avg.aggregate() #average value
 
     """
 
-    def __init__(self) -> None:
-        super().__init__()
-        self.sum = None
-        self.not_nans = None
+    def __init__(self, ensure_cpu_group: bool = False) -> None:
+        """
+        Args:
+            ensure_cpu_group: applies only to multi-gpu distributed setup. Optionally request to operate on CPU,
+                even if the default DistributedDataParallel group is on GPU. This will create an additional
+                CPU group with GLOO backend to reduce across processes on CPU (and maintain all intermediate data on CPU).
+                Defaults to False: use the default process group created during DistributedDataParallel initialization.
+        """
+
+        self.is_distributed = dist.is_available() and dist.is_initialized()
+        self.group = None
+        self.device = None
+
+        if self.is_distributed:
+            dist_cuda = dist.get_backend() == dist.Backend.NCCL
+            if ensure_cpu_group and dist_cuda:
+                self.group = dist.new_group(backend="gloo")  # create a new cpu group
```

Comment:
no sure if this class should manage the process group, this should be decoupled. also `new_group` takes additional parameters, it's not easy to make a suitable assumption about the system environment and have a robust implementation within this class

Reply:
@wyli good point, thank you. I've updated the code,  to have the process group as an option (if a user wants to provide a custom group).